tool/tsh/kube.go

```diff
@@ -270,6 +270,90 @@ func fetchKubeClusters(ctx context.Context, tc *client.TeleportClient) (teleport
 	return teleportCluster, kubeClusters, nil
 }
 
+// kubernetesStatus holds teleport client information necessary to populate the user's kubeconfig.
+type kubernetesStatus struct {
+	clusterAddr         string
+	teleportClusterName string
+	kubeClusters        []string
+	credentials         *client.Key
+}
+
+// fetchKubeStatus returns a kubernetesStatus populated from the given TeleportClient.
+func fetchKubeStatus(ctx context.Context, tc *client.TeleportClient) (*kubernetesStatus, error) {
+	var err error
+	kubeStatus := &kubernetesStatus{
+		clusterAddr: tc.KubeClusterAddr(),
+	}
+	kubeStatus.credentials, err = tc.LocalAgent().GetCoreKey()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	kubeStatus.teleportClusterName, kubeStatus.kubeClusters, err = fetchKubeClusters(ctx, tc)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return kubeStatus, nil
+}
+
+// buildKubeConfigUpdate returns a kubeconfig.Values suitable for updating the user's kubeconfig
+// based on the CLI parameters and the given kubernetesStatus.
+func buildKubeConfigUpdate(cf *CLIConf, kubeStatus *kubernetesStatus) (*kubeconfig.Values, error) {
+	v := &kubeconfig.Values{
+		ClusterAddr:         kubeStatus.clusterAddr,
+		TeleportClusterName: kubeStatus.teleportClusterName,
+		Credentials:         kubeStatus.credentials,
+	}
+
+	if cf.executablePath != "" {
```

Comment:
Nit: I would invert the condition to get rid of indentation

```go
if cf.executablePath == "" {
  return v
}
...
```

Reply:
Updated in the latest commit, thanks for the review!